logic/src/commonMain/kotlin/com/wire/kalium/logic/data/id/QualifiedId.kt

```diff
@@ -18,6 +18,7 @@ data class QualifiedID(
 }
 
 const val VALUE_DOMAIN_SEPARATOR = "@"
+val FEDERATION_REGEX = Regex("[^@.]+@[^@.]+\\.[^@]+")
```

Comment:
Are we 100% sure all domains have a dot (`.`) on them?

Also, can we instead use raw strings? So it's easier to copy and paste the pure RegEx without needing to escape them. For example with the `\` inside of it.

```kotlin
val FEDERATION_REGEX = """[^@.]+@[^@.]+\.[^@]+""".toRegex()
```